pallets/domains/src/types.rs

```diff
@@ -0,0 +1,85 @@
+use frame_support::pallet_prelude::*;
+use frame_support::traits::Currency;
+use sp_runtime::traits::Zero;
+
+use pallet_parachain_utils::{WhoAndWhenOf, new_who_and_when};
+
+use super::*;
+
+pub(crate) const IS_DOT_CHAR: fn(&u8) -> bool = |c: &u8| *c == b'.';
```

Comment:
What is the benefit of using this instead of function

Reply:
We reuse it a lot. And the common use-case is in `Filter`, where we provide a predicate, which is actually this const.